### DISCUSSION TITLE
Is there a way to have a layout that doesn't get matched against?

### DISCUSSION THREAD

**Discussion by mattmacdonald-m42** [kiliman/remix-flat-routes#60] [Q&A] [answered]
Hello!

Been using `remix-flat-routes` for a while and enjoying it. However I've run into something today that I'm unsure of a good solution to.

I have the following folder structure:
![Screenshot 2023-07-14 at 5 38 02 PM](https://github.com/kiliman/remix-flat-routes/assets/116587875/2bc41fb0-2dad-48bc-acb2-4a50a984d6be)

I added a layout to the `/signup` parent route since many of the underlying routes have the same structure, and they share a stepper to describe the steps. But ideally what I want is for `/signup` itself to be an invalid route, and only allow users to hit one of the pages that will trigger the full render. For example `/signup/intro`. Currently hitting `/signup` will render the template without any of the content, which makes sense since `_layout.tsx` is acting as `index.tsx` from my understanding.

My question is: is there a way to maintain the layout without allowing the user to hit `/signup`? Worst case I can provide a redirect in the `_layout` loader based on the path, but I was curious if there is a way to just make this not valid, but keep the layout for child routes.

**Comment by kiliman** [comment, marked as answer]
Hi great question. Typically if you want a shared layout without making it an actual route, you'd add a _pathless_ layout as a parent route. 

```
_shared/
  _layout.tsx
  signup/
    account-link/
    ...
```

Perhaps I could add a convention that would let you add a prefix like `!` to say wrap this with a pathless layout. 

`signup/!_layout.tsx` 

That might be tricky since I would have to check for that before processing any child routes since they would have to be nested under the _fake_ pathless layout. 

Anyway I hope I answered your question.

**Reply by mattmacdonald-m42** [reply]
Thanks for the answer. Yeah, I do have a "global" layout which follows the convention you mentioned above. Interesting thought about the `!` convention. I'm unfamiliar with the implementation of how paths are built, but would it be easier to add the prefix convention to the folder? Then it could render the layout and maybe just append the route to any children without setting it on the parent. Then for flat files it becomes a little bit easier since you can just check the filename in a similar fashion. Curious though, I might tinker around with it at some point as well and see if I get something that might be helpful enough for a request.

Anywho, thanks for your answer, it does answer my question. Ultimately I ended up just adding a redirect if the path ends in `/signup`, and I might just end up changing things around so that `/signup` ends up being the "intro" sub-view.
